### ISSUE TITLE
Fix state changes on unmounted components

### ISSUE THREAD

**Issue by asoomar** [asoomar/salad-bowl-app#42]
After improving the performance of displaying ads, a new issue arose where components would try to change their state even though they were not mounted. I think this is because if an ad is loaded but not shown, then you go to another page and load another ad, it triggers the previous ad's events (which are linked to that previous page's state even though it has already unmounted).

Action Steps:
- Need to figure out how to cancel all async calls before unmounting component
- Potentially have a flag before all the state changes in the listeners to ensure the component is mounted

Useful search result: [Here](https://www.google.com/search?q=cancel+async+call+in+componentwillunmount&oq=cancel+async+calls+compone&aqs=chrome.1.69i57j0.9818j0j4&sourceid=chrome&ie=UTF-8)

**Comment by asoomar** [comment]
This issue was fixed by using `AdMobInterstitial.removeAllListeners()` instead of `AdMobInterstitial.removeEventListener('interstitialDidLoad')` for each event. Previously, I tried to use the first approach but an error occurred where I could not remove event listeners that I didn't set up. This is why I took the second approach. 

However, I figured out that the event listeners weren't ever being removed on component unmount for some reason. I decided to just use all the event listeners (2 more than what I was already using) so that I could use `AdMobInterstitial.removeAllListeners()`. This method actually works as expected, and it even works when I load an ad on one component then switch to another component with an ad.
